src/python/dxpy/scripts/dx.py

```diff
@@ -1821,6 +1825,44 @@ def get_workflow(entity_result, args):
     describe_output = entity_result['describe']
     dump_executable(dx_obj, output_path, omit_resources=True, describe_output=describe_output)
 
+def do_debug(msg):
+    logging.debug(msg)
+
+def get_database(entity_result, args):
+    do_debug("dx.py#get_database - entity_result = {}".format(entity_result))
+    do_debug("dx.py#get_database - args = {}".format(args))
+    obj_id = entity_result['id']
+    project = entity_result['describe']['project']
+    do_debug("dx.py#get_database - project = {}".format(project))
+    output_path = get_output_path(obj_id,
+                                  entity_result['describe']['class'],
+                                  args)
+    do_debug("dx.py#get_database - output_path = {}".format(output_path))
+    from dxpy.utils.executable_unbuilder import dump_executable
+    print("Downloading database files", file=sys.stderr)
+    dx_obj = dxpy.DXDatabase(obj_id)
+    describe_output = entity_result['describe']
+    do_debug("dx.py#get_database - dx_obj = {}".format(dx_obj))
+
+    # Call /database-xxx/listFolder to fetch database file metadata
+    list_folder_args = {"folder": args.filename, "recurse": args.recurse}
```

Comment:
do we have a limit on the recurse or could the full database be downloaded?

Reply:
Full database download is permitted. Shall I add a 'maxfilecount' argument to constrain this? The only downside I see is that it's kind of arbitrary, for example if you use maxfilecount of 1000 and there are 5000 files, which 1000 should be included...it would be in the order of the list returned by s3.listObjects.

Reply:
hmm, how about a warning and only allow full database download if a "-f" argument is supplied. i am ok with downloading the whole thing since this a dnanexus only feature but don't want it to happen unknowingly..what do you think? @durae your thoughts?

Reply:
I would suggest a flag too (so it doesn't need to be run interactively), but we can't use `-f`. That's already reserved for `--overwrite`.

Reply:
@durae and @arun, done. There is an --allow-all-files argument now which you need to specify if you omit the --filename arg. Otherwise, you get an error.